src/ftest.jl

```diff
@@ -0,0 +1,75 @@
+type FTestResult
+    mod1::RegressionModel
+    mod2::RegressionModel
+    SSR1::Float64 # sum of squared residuals, model 1
+    SSR2::Float64 # sum of squared residuals, model 2
+    df1::Int # degrees of freedom, model 1 residuals
+    df2::Int # degrees of freedom, model 2 residuals
+    MSR1::Float64 # mean of squared residuals, model 1
+    MSR2::Float64 # mean of squared residuals, model 2
+    fstat::Float64 # f statistic
+    pval::Float64 # p value
+end
+
+type MultiFTestResult
+    results::Array{FTestResult, 1}
+end
+
+function ftest(mod1::RegressionModel, mod2::RegressionModel)
```

Comment:
The F test is only valid for linear regressions, so the signature should be stricter.